snapshot-jdbc-test/src/main/java/com/hazelcast/jet/tests/snapshot/jdbc/JdbcSinkVerifier.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.tests.snapshot.jdbc;
+
+import com.hazelcast.logging.ILogger;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.PriorityQueue;
+import javax.sql.DataSource;
+
+import static com.hazelcast.jet.impl.util.Util.uncheckRun;
+import static com.hazelcast.jet.tests.snapshot.jdbc.JdbcSinkTest.DataSourceSupplier.getDataSourceSupplier;
+import static com.hazelcast.jet.tests.snapshot.jdbc.JdbcSinkTest.TABLE_PREFIX;
+
+public class JdbcSinkVerifier {
+
+    private static final int SLEEP_AFTER_VERIFICATION_CYCLE_MS = 1000;
+    private static final int ALLOWED_NO_INPUT_MS = 600000;
+    private static final int QUEUE_SIZE_LIMIT = 20_000;
+    private static final int PRINT_LOG_ITEMS = 10_000;
+
+    private final String tableName;
+
+    private final Thread consumerThread;
+    private final String name;
+    private final ILogger logger;
+    private final String connectionUrl;
+
+    private volatile boolean finished;
+    private volatile Throwable error;
+    private long counter;
+    private final PriorityQueue<Integer> verificationQueue = new PriorityQueue<>();
+
+    public JdbcSinkVerifier(String name, ILogger logger, String connectionUrl) {
```

Comment:
I also see some similarities between JmsSinkVerifier, GeneratedFilesVerifier and JdbcSinkVerifier. some of the logic can be centralized